### PR TITLE
Linking with Stan.jl

### PR DESCRIPTION
**Pull request by Ayush-iitkgp**
This PR is the firt step to integrate with Stan.jl. 
Few future improvements include:
1. Allowing users to input the priors using Distributions.jl (the defualt behaviour is that the parameters follow Normal(0,1)).
2. Output more readable output such as the mean and the quantiles of the parameters. At present, it returns the Mamba chains

### CODE REVIEW

**Comment by ChrisRackauckas** [comment]
Looks good. Can it get some unit tests to make sure the pieces are producing the right strings, and that this generates some chains without errors?

**Comment by Ayush-iitkgp** [comment]
@gabrielgellner @goedman please have a look. Your comments are valuable.

File: src/bayesian_inference.jl [hunk 1]
```diff
@@ -27,11 +27,16 @@ function generate_priors(f)
...
 end
 
-function bayesian_inference(prob::DEProblem,t,data;alg=:integrate_ode_rk45,num_samples=1, num_warmup=1,kwargs...)
+function bayesian_inference(prob::DEProblem,t,data;alg=:rk45,num_samples=1, num_warmup=1,kwargs...)
   length_of_y = string(length(prob.u0))
   f = prob.f
   length_of_parameter = string(length(f.params))
-  alg = string(alg)
+  if alg ==:rk45
+  alg = integrate_ode_rk45
```
Comment (ChrisRackauckas) [inline src/bayesian_inference.jl]: don't these need to be strings?

**Comment by Ayush-iitkgp** [comment]
A slight change in API, now the users can pass distribution of the priors using the Normal distribution from Distributions.jl.

**Comment by ChrisRackauckas** [comment]
I don't think the default `Normal(0,1)` makes sense. Priors should always be required.

File: src/bayesian_inference.jl [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+    real x_r[0];
+    int x_i[0];
+  }
+  parameters {
+    vector<lower=0>[$length_of_y] sigma;
+    real theta[$length_of_parameter];
+  }
+  model{
+    real u_hat[T,$length_of_y];
+    sigma ~ inv_gamma(2, 3);
```
Comment (ChrisRackauckas) [inline src/bayesian_inference.jl]: This should be set when the user chooses the likelihood somehow.

File: src/bayesian_inference.jl [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+    $priors_string
+    u_hat = $algorithm(sho, u0, t0, ts, theta, x_r, x_i);
+    for (t in 1:T){
+      u[t] ~ normal(u_hat[t], sigma);
```
Comment (ChrisRackauckas) [inline src/bayesian_inference.jl]: We should allow the choice here somehow. Maybe in a future PR.

File: src/bayesian_inference.jl (hunk 2 above)
Comment (Ayush-iitkgp) [reply src/bayesian_inference.jl]: This is the default nature. Should ask users to input the distribution of sigma as well?

File: src/bayesian_inference.jl (hunk 3 above)
Comment (Ayush-iitkgp) [reply src/bayesian_inference.jl]: Very much possible :)
Let's get the Normal Distribution working.

File: src/bayesian_inference.jl (hunk 2 above)
Comment (ChrisRackauckas) [reply src/bayesian_inference.jl]: For now, yes. Later this should be part of the likelihood choice I think.

Comment (ChrisRackauckas) [reply src/bayesian_inference.jl]: For now, keep this I guess. It'll get addressed when likelihoods are addressed. General likelihoods should have different ways of setting hyperparameters.

**Comment by Ayush-iitkgp** [comment]
The tests are passing on my Mac.

**Comment by goedman** [comment]
I’ll continue looking at the results tomorrow, but looking at the diagnostic plots doesn’t look promising. Also the tests too often fail on my machine, also a Mac.

The number of samples in runtests.jl is also rather low.

But I will look at it again tomorrow.

**Comment by ChrisRackauckas** [comment]
Yeah, it might need to be run longer, but that means we might need to have CI just on Linux so that way it doesn't bog down our CI times. I made those changes, and we should check in more detail what's going on. My guess is it has to do with the fact that negative parameters don't make sense, so I know Turing.jl needed the truncated priors to do well.

**Pull request merged**